plugins/org.eclipse.fordiac.ide.fbtypeeditor/src/org/eclipse/fordiac/ide/fbtypeeditor/editparts/FBTypeEditPart.java

```diff
@@ -38,38 +39,34 @@
 import org.eclipse.jface.resource.JFaceResources;
 import org.eclipse.jface.util.IPropertyChangeListener;
 import org.eclipse.swt.events.ControlListener;
-import org.eclipse.swt.widgets.Display;
 
 public class FBTypeEditPart extends AbstractConnectableEditPart {
 
 	private ControlListener controlListener;
 	final List<AbstractContainerElement> containerChildren = new ArrayList<>(6);
 	private DiagramFontChangeListener fontChangeListener;
 
-	private final Adapter adapter = new EContentAdapter() {
+	private final Adapter versionInfoAdapter = new AdapterImpl() {
 		@Override
 		public void notifyChanged(final Notification notification) {
 			super.notifyChanged(notification);
-			if (Notification.REMOVING_ADAPTER != notification.getEventType()) {
-				final Object feature = notification.getFeature();
-				if ((LibraryElementPackage.eINSTANCE.getVersionInfo().equals(feature))
-						|| (LibraryElementPackage.eINSTANCE.getVersionInfo_Version().equals(feature))) {
-					getFigure().updateVersionInfoLabel();
-				}
-
-				Display.getDefault().syncExec(() -> {
-					if ((null != getParent()) && (null != getFigure()) && (getFigure().isShowing())) {
-						refresh();
-					}
-				});
+			final Object feature = notification.getFeature();
+			if (notification.getEventType() == Notification.ADD
+					&& LibraryElementPackage.eINSTANCE.getLibraryElement_VersionInfo().equals(feature)) {
+				((EObject) notification.getNewValue()).eAdapters().add(this);
+				getFigure().updateVersionInfoLabel();
+			} else if (Notification.REMOVING_ADAPTER != notification.getEventType()
```

Comment:
I would suggest using `!Notification.isTouch()` instead to filter for actual changes.